common/configuration/class.FileSystemComponent.php

```diff
@@ -244,38 +244,54 @@ public function check()
         return $returnValue;
     }
 
-    /**
-     * If file is readable.
-     *
-     * @access public
-     * @author Aleh Hutnikau, <hutnikau@1pt.com>
-     * @param string $location File location
-     * @return boolean
-     */
-    public function isReadable($location = null)
+    private function hasLocationAccess($location = null, $rule = 'Readable')

```

Comment:
Could you check if $rule parameter is valid. 

> in_array($rule, array('writable', 'readable', 'executable'))

And with PHP doc block, it should be perfect